bmorph/util/mizuroute_utils.py

```diff
@@ -0,0 +1,217 @@
+        else:
+            is_gauge.append(False)
+
+    routed['is_gauge'].values[:]  = is_gauge
+    
+    routed['distance_to_up_gauge'] = 0 * routed['is_headwaters']
+    routed['distance_to_down_gauge'] = 0 * routed['is_headwaters']
+    routed['cdf_blend_factor'] = 0 * routed['is_headwaters']
+    routed['up_seg'] = 0 * routed['is_headwaters']
+    routed['upstream_ref_seg'] = np.nan * routed['is_headwaters']
+    routed['downstream_ref_seg'] = np.nan * routed['is_headwaters']
+    
+    routed['up_seg'].values = [find_upstream(routed, s) for s in routed['seg'].values]
+    routed['distance_to_up_gauge'].values =[walk_upstream(routed, s) for s in routed['seg']]
+    routed['distance_to_down_gauge'].values = [walk_downstream(routed, s) for s in routed['seg']]
+    routed['cdf_blend_factor'].values = (routed['distance_to_up_gauge'] 
+                                         / (routed['distance_to_up_gauge'] + routed['distance_to_down_gauge'])).values
+    routed['cdf_blend_factor'] = routed['cdf_blend_factor'].where(~np.isnan(routed['cdf_blend_factor']), other=0.0)
+    
+    for site, seg in site_2_seg.items():
+        routed['downstream_ref_seg'].loc[{'seg':seg}] = seg
+        routed['upstream_ref_seg'].loc[{'seg':seg}] = seg
+    
+    for seg in routed['seg'].values:
+        if seg in seg_2_site.keys() or routed.sel(seg=seg)['down_seg'].values[()] not in routed['seg'].values:
+            continue
+        current_seg = seg
+        while routed.sel(seg=current_seg)['down_seg'].values[()] not in seg_2_site.keys():
+            current_seg = routed.sel(seg=current_seg)['down_seg'].values[()]
+        downstream_gauge_seg = routed.sel(seg=current_seg)['down_seg'].values[()]
+        routed['downstream_ref_seg'].loc[{'seg': seg}] = downstream_gauge_seg
+        
+    for seg in routed['seg'].values:
+        if seg in seg_2_site.keys() or routed.sel(seg=seg)['up_seg'].values[()] not in routed['seg'].values:
+            continue
+        current_seg = seg
+        while routed.sel(seg=current_seg)['up_seg'].values[()] not in seg_2_site.keys():
+            current_seg = routed.sel(seg=current_seg)['up_seg'].values[()]
+            if current_seg is None:
+                break
+        if current_seg is None:
+            routed['upstream_ref_seg'].loc[{'seg':seg}] = routed['downstream_ref_seg'].loc[{'seg': seg}]
+        else:
+            upstream_gauge_seg = routed.sel(seg=current_seg)['up_seg'].values[()]
+            routed['upstream_ref_seg'].loc[{'seg': seg}] = upstream_gauge_seg
+        
+    routed['downstream_ref_seg'] = (routed['downstream_ref_seg'].where(
+        ~np.isnan(routed['downstream_ref_seg']), other=routed['upstream_ref_seg']))
+    routed['upstream_ref_seg'] = (routed['upstream_ref_seg'].where(
+        ~np.isnan(routed['upstream_ref_seg']), other=routed['downstream_ref_seg']))
+    
+    return routed
+    
+def annotate_variable(routed: xr.Dataset, condition: xr.Dataset, var_label: str, condition_var: str, 
```

Comment:
As with the other one, the `annotate` naming isn't very descriptive to me. Perhaps this one could be something like `map_var_to_reference_segs`?

Reply:
changed!